src/test/java/htsjdk/samtools/TbiEqualityChecker.java

```diff
@@ -0,0 +1,236 @@
+    for (int i = 0; i <= bins1.maxBinNumber; i++) {
+      assertEquals(bins1.getBin(i), bins2.getBin(i), identical || i != bins1.maxBinNumber);
+    }
+  }
+
+  private void assertEquals(Bin bin1, Bin bin2, boolean identical) throws IOException {
+    if (bin1 == null || bin2 == null) {
+      Assert.assertEquals(bin1, bin2);
+      return;
+    }
+    Assert.assertEquals("Bin number", bin1.getBinNumber(), bin2.getBinNumber());
+    Assert.assertEquals("Chunk list size", bin1.getChunkList().size(), bin2.getChunkList().size());
+    for (int i = 0; i < bin1.getChunkList().size(); i++) {
+      assertEquals(bin1.getChunkList().get(i), bin2.getChunkList().get(i), identical);
+    }
+  }
+
+  private void assertEquals(LinearIndex linearIndex1, LinearIndex linearIndex2) {
+    Assert.assertEquals(
+        "Linear index ref",
+        linearIndex1.getReferenceSequence(),
+        linearIndex2.getReferenceSequence());
+    Assert.assertEquals("Linear index size", linearIndex1.size(), linearIndex2.size());
+    Assert.assertEquals(
+        "Linear index start", linearIndex1.getIndexStart(), linearIndex2.getIndexStart());
+    assertArrayEquals(
+        "Linear index entries", linearIndex1.getIndexEntries(), linearIndex2.getIndexEntries());
+  }
+
+  private void assertEquals(Chunk chunk1, Chunk chunk2, boolean identical) {
+    if (identical) {
+      Assert.assertEquals("Chunk start", chunk1.getChunkStart(), chunk2.getChunkStart());
+      Assert.assertEquals("Chunk end", chunk1.getChunkEnd(), chunk2.getChunkEnd());
+    } else {
+      assertEquivalent("Chunk start", chunk1.getChunkStart(), chunk2.getChunkStart());
+      assertEquivalent("Chunk end", chunk1.getChunkEnd(), chunk2.getChunkEnd());
+    }
+  }
+
+  private void assertEquivalent(
+      String message, long virtualFilePointer1, long virtualFilePointer2) {
+    // seek to the given positions check they are actually equivalent
+    long norm1 = normalizeVirtualFilePointer(virtualFilePointer1);
+    long norm2 = normalizeVirtualFilePointer(virtualFilePointer2);
+    Assert.assertEquals(message, norm1, norm2);
+  }
+
+  private long normalizeVirtualFilePointer(long virtualFilePointer) {
+    try {
+      blockStream.seek(virtualFilePointer);
+      return blockStream.getFilePointer();
+    } catch (IOException e) {
+      fail("Failed to seek to " + BlockCompressedFilePointerUtil.asString(virtualFilePointer));
+      return -1; // never reached
+    }
+  }
+
+  private void dump(BinningIndexContent binningIndexContent) {
+    PrintWriter pw = new PrintWriter(System.out);
+    writeReference(pw, binningIndexContent);
+  }
+
+  public static void dump(PrintWriter pw, TabixIndex tbi) {
+    BinningIndexContent[] binningIndexContents = TabixIndexMerger.getBinningIndexContents(tbi);
+    for (BinningIndexContent content : binningIndexContents) {
+      writeReference(pw, content);
+    }
+  }
+
+  public static void writeReference(PrintWriter pw, final BinningIndexContent content) {
```

Comment:
It looks like big chunks of this are shared with TabixPlainText. It would be good to deduplicate them.